crates/viewer/re_space_view_dataframe/src/view_query_v2/blueprint_io.rs

```diff
@@ -0,0 +1,235 @@
+    }
+
+    pub(super) fn save_filter_by_event(
+        &self,
+        ctx: &ViewerContext<'_>,
+        filter_by_event: &components::FilterByEvent,
+    ) {
+        self.query_property
+            .save_blueprint_component(ctx, filter_by_event);
+    }
+
+    pub(crate) fn latest_at_enabled(&self) -> Result<bool, SpaceViewSystemExecutionError> {
+        Ok(self
+            .query_property
+            .component_or_empty::<components::ApplyLatestAt>()?
+            .map_or(false, |comp| *comp.0))
+    }
+
+    pub(crate) fn save_latest_at_enabled(&self, ctx: &ViewerContext<'_>, enabled: bool) {
+        self.query_property
+            .save_blueprint_component(ctx, &components::ApplyLatestAt(enabled.into()));
+    }
+
+    pub(super) fn save_selected_columns(
+        &self,
+        ctx: &ViewerContext<'_>,
+        columns: impl IntoIterator<Item = ColumnSelector>,
+    ) {
+        let mut selected_columns = datatypes::SelectedColumns::default();
+        for column in columns {
+            match column {
+                ColumnSelector::Control(_) => {}
+                ColumnSelector::Time(desc) => {
+                    selected_columns
+                        .time_columns
+                        .push(desc.timeline.as_str().into());
+                }
+                ColumnSelector::Component(desc) => {
+                    let blueprint_component_descriptor =
+                        datatypes::ComponentColumnSelector::new(&desc.entity_path, desc.component);
+
+                    selected_columns
+                        .component_columns
+                        .push(blueprint_component_descriptor);
+                }
+            }
+        }
+
+        self.query_property
+            .save_blueprint_component(ctx, &components::SelectedColumns(selected_columns));
+    }
+
+    pub(super) fn save_all_columns_selected(&self, ctx: &ViewerContext<'_>) {
+        self.query_property
+            .clear_blueprint_component::<components::SelectedColumns>(ctx);
+    }
+
+    pub(super) fn save_all_columns_unselected(&self, ctx: &ViewerContext<'_>) {
+        self.query_property
+            .save_blueprint_component(ctx, &components::SelectedColumns::default());
+    }
+
+    /// Given a schema, list the columns that should be visible, according to the blueprint.
+    pub(crate) fn apply_column_visibility_to_schema(
+        &self,
+        ctx: &ViewerContext<'_>,
+        schema: &[ColumnDescriptor],
+    ) -> Result<Option<Vec<ColumnSelector>>, SpaceViewSystemExecutionError> {
+        let selected_columns = self
+            .query_property
+            .component_or_empty::<components::SelectedColumns>()?;
+
+        // no selected columns means all columns are visible
+        let Some(datatypes::SelectedColumns {
+            time_columns,
+            component_columns,
+        }) = selected_columns.as_deref()
+        else {
+            return Ok(None);
```

Comment:
I figure this means `None` == all columns? Would be nice to document that in the doc-comment